config_explorer/src/config_explorer/capacity_planner.py

```
@@ -15,92 +17,152 @@ def get_model_info_from_hf(model_name: str, hf_token: str | None = None):
+        "I32": 4,
+        "INT32": 4,
+        "I16": 2,
+        "INT16": 2,
+        "I8": 1,
+        "INT8": 1,
+        "U8": 1,
+        "U4": 0.5,
+        "I4": 0.5,
+        "INT4": 0.5,
 
-# GPU and KV cache
-def min_gpu_req(model_info: ModelInfo, gpu_memory: int) -> int:
+        # Boolean
+        "BOOL": 1,  # stored as byte per element
+    }
+
+    if precision in mapping:
+        return mapping[precision]
+    else:
+        # Try to infer the precision from the first whole number
+        match = re.search(r"\d+", precision)
+        if match:
+            bits = int(match.group(0))
+            if bits % 8 == 0:
+                return bits // 8
+
+    # Return BF16's precision as last resort
+    return 2
+
+def parameter_memory_req(parameter: int, precision: str) -> int:
     """
-    Calculates the minimum GPU count needed for the model
+    Calculates the memory requirement for the number of parameters for the specified precision
     """
 
-    model_memory_gb = model_memory_req(model_info)
-    return math.ceil(model_memory_gb / gpu_memory)
+    precision_byte = precision_to_byte(precision)
+    return parameter * precision_byte / (1024 ** 3)
 
-def get_model_config_from_hf(model_name: str, hf_token: str=None) -> AutoConfig:
+def model_memory_req(model_info: ModelInfo) -> int:
     """
-    Returns LLM model config
+    Calculates the GPU memory required for loading the model
     """
+    try:
+        model_params = model_info.safetensors.parameters
+        memory = 0
+        for precision, num_params in model_params.items():
+            memory += parameter_memory_req(num_params, precision)
 
-    model_config = AutoConfig.from_pretrained(
-        model_name,
-        trust_remote_code=True,
-        token=hf_token or None,
-    )
+        return memory
 
-    # For LLMs
-    if hasattr(model_config, "text_config"):
-        model_config = model_config.text_config
+    except Exception as e:
+        print(e)
+        return -1
```

What errors are you typically catching here?

---

Wouldn't it be more appropriate to have an attribute or not implemented error here (in the library)? If this breaks the Streamlit demo UI, then have a try/except there to alert the user that model isn't supported yet.

---

Handled in the UI client.

---

I'm suggesting that in this library, don't have this `except Exception...` catch-all (bad practice). Let the exception raise, but handle within the UI if needed.